h/services/annotation_json_presentation.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from sqlalchemy.orm import subqueryload
+
+from memex import resources
+from memex.interfaces import IGroupService
+
+from h import formatters
+from h import models
+from h import presenters
+from h import storage
+
+
+class AnnotationJSONPresentationService(object):
+    def __init__(self, session, authenticated_user, group_svc, links_svc):
+        self.session = session
+        self.group_svc = group_svc
+        self.links_svc = links_svc
+
+        self.formatters = [
+            formatters.AnnotationFlagFormatter(self.session, authenticated_user)
+        ]
+
+    def present(self, annotation_resource):
+        presenter = self.get_presenter(annotation_resource)
+        return presenter.asdict()
+
+    def present_all(self, annotation_ids):
+        def eager_load_documents(query):
+            return query.options(
+                subqueryload(models.Annotation.document))
+
+        annotations = storage.fetch_ordered_annotations(
+            self.session, annotation_ids, query_processor=eager_load_documents)
+
+        # preload formatters, so they can optimize database access
+        for formatter in self.formatters:
+            formatter.preload(annotation_ids)
+
+        return [self.present(
+                    resources.AnnotationResource(ann, self.group_svc, self.links_svc))
+                for ann in annotations]
+
+    def get_presenter(self, annotation_resource):
```

Comment:
I guess there could be other use cases here, but I’m not quite sure what they would be, since we’ve covered the cases for “present a single thing” and “present a whole bunch of things”. To me, the fact that we can instantiate a single presenter object for an annotation is an implementation detail, and one that we could quite conceivably end up swapping out at a later date. From the perspective of what’s currently calling the service, the service *is* what’s doing the presenting.

In most places, I’d agree that we can get away without a hard split between private and public methods. In this case, though, it’s a service, so I think there’s more benefit to making it extra clear what’s in the service’s interface and what isn’t.